### PR TITLE
Changes to allow for summary action bars to be declared inside tabs

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
Also, ensures that switching two action bars on and off works.

### CODE REVIEW

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.ts [hunk 1]
```diff
@@ -137,7 +163,52 @@ export class SkySummaryActionBarComponent implements AfterViewInit, OnDestroy {
...
+    if (!this.observer) {
+      this.observer = this.observerService.create((mutations: MutationRecord[]) => {
+        if ((<HTMLElement>mutations[0].target).attributes.getNamedItem('hidden')) {
+          this.adapterService.revertBodyElementStyles();
+          this.removeResizeListener();
+        } else {
+          setTimeout(() => {
+            this.adapterService.styleBodyElementForActionBar(this.elementRef);
+            this.setupResizeListener();
+          });
```
Comment (blackbaud-conorwright) [inline src/app/public/modules/summary-action-bar/summary-action-bar.component.ts]: Why is this in a setTimeout? Were there issues with it running within the lifecycle?

File: src/app/public/modules/summary-action-bar/summary-action-bar-adapter.service.ts [hunk 2]
```diff
@@ -79,16 +56,12 @@ export class SkySummaryActionBarAdapterService implements OnDestroy {
           el = el.parentElement;
         }
         return SkySummaryActionBarType.StandardModal;
+      } else if (el.classList.contains('sky-tab')) {
```
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: Unfortunately no. To determine the type of action bar we have to traverse to determine if there is a tab or modal parent.

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/summary-action-bar/summary-action-bar.component.ts]: Ya. The observer fires when the tab that the action bar is in either is shown or hidden. However, if the user is switching from one tab that has an action bar to another that has a different action bar then without the `setTimeout` It is possible that the new one will run its setup before the old one is removed and the removal of the old one will cause the body styles to be removed. 

The timing here is up to Angular so there isn't much we can do about it but the `setTimeout` ensures that the new action bar is always added after the removal of the old one.

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.ts [hunk 3]
```diff
@@ -91,13 +112,18 @@ export class SkySummaryActionBarComponent implements AfterViewInit, OnDestroy {
   public ngOnDestroy(): void {
     if (!(this.type === SkySummaryActionBarType.StandardModal ||
       this.type === SkySummaryActionBarType.FullPageModal)) {
-        this.adapterService.revertBodyElementStyles();
-        this.adapterService.removeResizeListener();
+      this.adapterService.revertBodyElementStyles();
+      this.removeResizeListener();
+
+      if (this.type === SkySummaryActionBarType.Tab) {
```
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/summary-action-bar/summary-action-bar.component.ts]: Good call. Removed.

File: src/app/public/modules/summary-action-bar/summary-action-bar-adapter.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/summary-action-bar/summary-action-bar-adapter.service.ts]: dang, oh well

**Comment by Blackbaud-TrevorBurch** [comment]
> I'm seeing an issue when I launch this in IE11 and Edge. If I shorten the window to a width that causes the buttons to collapse and the action bar to be collapsible. Then clicking the [...] hamburger button or the collapse chevron do not do anything

I'm not seeing this behavior...... I'll reach out one on one to see what you're seeing

**Comment by Blackbaud-TrevorBurch** [comment]
> I'm not seeing this behavior...... I'll reach out one on one to see what you're seeing

Never mind. This is only happening in tabs.... should have known that. I see it now.

File: src/app/public/modules/summary-action-bar/summary-action-bar.component.ts [hunk 4]
```diff
@@ -177,12 +177,12 @@ export class SkySummaryActionBarComponent implements AfterViewInit, OnDestroy {
...
 
-    const config = { attribute: true, attributeFilter: ['hidden'], childList: false, characterDate: false };
+    const config = { attributes: true, attributeFilter: ['hidden'], childList: false, characterDate: false };
     let el = this.elementRef.nativeElement;
     do {
       if (el.classList.contains('sky-tab')) {
-        this.observer.observe(el, config);
-      }
+         this.observer.observe(el, config);
+       }
```
Comment (blackbaud-conorwright) [inline src/app/public/modules/summary-action-bar/summary-action-bar.component.ts]: Why are these spaced over?

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/summary-action-bar/summary-action-bar.component.ts]: Honestly not sure. May have been from some commenting I removed. I've fixed it.

**Pull request merged**